### PR TITLE
feat(extension): add storytelling layer

### PR DESCRIPTION
**Pull request by airslice**
## Overview

This PR adds support of storytelling layer.

- Create story from toolbar.
- Select story layer to show:
  - Story edit section when in story mode.
  - Story inspect section when in other modes.
- Use capture editor to create / edit capture.
  - Edit with markdown.
 - Select story object (capture) to show capture inspector.
  - Support edit / re-capture when in story mode. 
 
This PR adds some ViewComponent as common components. They are not been used in other view UIs but maybe can be used if we need to refactor to alignment the UI in the future.

## Note

- This PR added `allowedEvents` to ScreenSpaceSelectionHandler since story mode needs only point selecion.
- This PR added a `preventToolKeyDownAtom` to prevent the keypress event when needed (eg. when user input text).

## Screen shots

localhost_3000_(Desktop 1920)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (21)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (22)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (23)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (24)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (25)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (26)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (27)
localhost_3000_scene_01hnyhekfpzjhh81qhg61vveg5_widgets(Desktop 1920) (28)

### CODE REVIEW

File: extension/src/shared/reearth/hooks/useLayer.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const useLayer = ({
   useEffect(() => {
     const layerId = layerIdRef.current;
     if (!layerId || !loaded) return;
-
+
```
Comment (keiya01) [inline extension/src/shared/reearth/hooks/useLayer.ts:46]:
```suggestion
```

File: extension/src/shared/reearth/layers/story.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      type: "Story",
+      key: capture.id,
+    });
+
+    return {
+      type: "geojson",
+      value: {
+        type: "Feature",
+        properties: {
+          [STORY_MARKER_ID_PROPERTY]: objectId,
```
Comment (keiya01) [inline extension/src/shared/reearth/layers/story.ts:31]: Is it working? Should we generate the id for each marker...?

Comment (airslice) [reply extension/src/shared/reearth/layers/story.ts:31]: Yes this `StoryLayer` is actually marker layer.
So each capture will generate a objectId and been used in two reearth layers (one for balloon icon and one for camera icon). Maybe i can refactor some to make it looks more clear but basically it's okey i think.

File: extension/src/prototypes/view/selection/StoryCaptureContent.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      <Divider />
+      <SectionWrapper>
+        <Content>
+          <CaptureTitle>{capture?.title}</CaptureTitle>
+          <ViewMarkdownViewer content={capture?.content} />
+        </Content>
+        {editMode && (
+          <ButtonsWrapper>
+            <Button size="small" variant="outlined" fullWidth onClick={handleRecapture}>
+              Re-capture
```
Comment (keiya01) [inline extension/src/prototypes/view/selection/StoryCaptureContent.tsx:125]:
```suggestion
              再キャプチャ
```

File: extension/src/prototypes/view/selection/StoryCaptureContent.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            </Button>
+            <Button size="small" variant="outlined" fullWidth onClick={handleEdit}>
+              Edit
```
Comment (keiya01) [inline extension/src/prototypes/view/selection/StoryCaptureContent.tsx:128]:
```suggestion
              編集
```

File: extension/src/prototypes/view/selection/StoryEditSection.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        onCaptureRemove={handleRemoveCapture}
+      />
+      <ButtonWrapper>
+        <Button
+          size="small"
+          variant="outlined"
+          fullWidth
+          startIcon={<CameraIcon />}
+          onClick={handleOpenNewCapture}>
+          New Capture
```
Comment (keiya01) [inline extension/src/prototypes/view/selection/StoryEditSection.tsx:72]:
```suggestion
          新しいキャプチャ
```

File: extension/src/prototypes/view/selection/StoryInspectSection.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+                page={currentCaptureIndex + 1}
+                siblingCount={1}
+                boundaryCount={1}
+                onChange={handleChange}
+              />
+            </PaginationWrapper>
+          </>
+        )
+      ) : (
+        <NoCaptures>No captures.</NoCaptures>
```
Comment (keiya01) [inline extension/src/prototypes/view/selection/StoryInspectSection.tsx:98]:
```suggestion
        <NoCaptures>キャプチャがありません</NoCaptures>
```

File: extension/src/prototypes/view/ui-containers/ToolButtons.tsx [hunk 7]
```diff
@@ -55,6 +57,16 @@ export const ToolButtons: FC = () => {
     [send, setSketchType],
   );
 
+  const showCreateStory = useSetAtom(showCreateStoryAtom);
+  const handleCreateStory = useCallback(() => {
+    showCreateStory(true);
+  }, [showCreateStory]);
+
+  const storyItems = useMemo(
+    () => [{ title: "New Story", icon: <StoryIcon />, onClick: handleCreateStory }],
```
Comment (keiya01) [inline extension/src/prototypes/view/ui-containers/ToolButtons.tsx:66]:
```suggestion
    () => [{ title: "新しいストーリー", icon: <StoryIcon />, onClick: handleCreateStory }],
```

File: extension/src/shared/ui-components/story/CaptureListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        onClick: editCapture,
+      },
+      {
+        label: "Re-Capture",
+        onClick: recaptureCapture,
+      },
+      {
+        label: "Delete",
+        onClick: removeCapture,
+      },
```
Comment (keiya01) [inline extension/src/shared/ui-components/story/CaptureListItem.tsx:79]:
```suggestion
      {
        label: "視点",
        onClick: viewCapture,
      },
      {
        label: "編集",
        onClick: editCapture,
      },
      {
        label: "再キャプチャ",
        onClick: recaptureCapture,
      },
      {
        label: "削除",
        onClick: removeCapture,
      },
```

File: extension/src/shared/view/ui-container/story/StoryCaptureEditor.tsx [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  return (
+    <ViewDialog
+      open={open}
+      title="Capture Editor"
+      icon={<CameraIcon />}
+      disableSubmit={!title.trim() || !content.trim()}
+      onClose={onClose}
+      onSubmit={handleSubmit}>
+      <ViewContentColumn>
+        <ViewTextField placeholder="Title" value={title} onChange={handleTitleChange} />
```
Comment (keiya01) [inline extension/src/shared/view/ui-container/story/StoryCaptureEditor.tsx:61]:
```suggestion
        <ViewTextField placeholder="タイトル" value={title} onChange={handleTitleChange} />
```

File: extension/src/shared/view/ui-container/story/StoryCaptureEditor.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+  const preventToolKeyDown = useSetAtom(preventToolKeyDownAtom);
+  useEffect(() => {
+    preventToolKeyDown(open);
+  }, [open, preventToolKeyDown]);
+
+  return (
+    <ViewDialog
+      open={open}
+      title="Capture Editor"
```
Comment (keiya01) [inline extension/src/shared/view/ui-container/story/StoryCaptureEditor.tsx:55]:
```suggestion
      title="キャプチャエディタ"
```

**Review by keiya01 — APPROVED** [review]

**Pull request merged**